### PR TITLE
[php/ng] support the use of a list of php versions

### PR DESCRIPTION
**Pull request by arthurlogilab**
Related to #138

### CODE REVIEW

File: php/ng/ini.jinja [hunk 1]
```diff
@@ -10,5 +10,5 @@
     - source: salt://php/ng/files/php.ini
     - template: jinja
     - context:
-        config: {{ serialize(settings) }}
+        config: {{ serialize(odict(settings)) }}
```
Comment (arthurlogilab) [inline php/ng/ini.jinja]: not sure this is needed

File: php/ng/fpm/config.sls [hunk 2]
```diff
@@ -11,6 +11,39 @@
...
+    {% set conf = php.lookup.fpm.conf|replace(first_version, version) %}
+    {% set pools = php.lookup.fpm.pools|replace(first_version, version) %}
+
+    {% for key, value in conf_settings.items() %}
+      {% if value is string %}
+        {% do conf_settings.update({key: value.replace(first_version, version)}) %}
+      {% endif %}
+    {% endfor %}
+    {% do conf_settings.global.update({'pid': '/var/run/php' + version + '-fpm.pid' }) %}
+    {% do conf_settings.global.update({'error_log': '/var/log/php' + version + '-fpm.log' }) %}
```
Comment (arthurlogilab) [inline php/ng/fpm/config.sls]: not very satisfied with this part, I would want the loop above to be more generic but for some reason I couldn't get it to change the subdictionnary

**Comment by arthurlogilab** [comment]
the approach here is to "post-process the map dictionnary" when the php version is a list

File: php/ng/fpm/config.sls [hunk 3]
```diff
@@ -11,6 +11,39 @@
   {% endif %}
 {% endfor %}
 
+
+{% if salt['pillar.get']('php:ng:version') is iterable %}
```
Comment (arthurlogilab) [inline php/ng/fpm/config.sls]: this test needs to be changed since a string is also an iterable

File: php/ng/fpm/init.sls [hunk 4]
```diff
@@ -5,14 +5,34 @@ include:
   - php.ng.fpm.service
   - php.ng.fpm.pools
 
+{% set pillar_php_ng_version = salt['pillar.get']('php:ng:version', '7.0') %}
 extend:
   php_fpm_service:
     service:
       - watch:
+    {% if pillar_php_ng_version is iterable and pillar_php_ng_version is not string %}
```
Comment (arthurlogilab) [inline php/ng/fpm/init.sls]: now I'm not sure how this handles when 7.3 as a float is used. Is setting the version without quotes supported by the existing version of the formula ?

**Comment by pather87** [comment]
I currently see the issue, that you cannot remove the default www.conf with multi version use. 
In single version use, you could add a php:ng:fpm:pools:www.conf:enabled: False and it would remove the default pool for you.
Since we use a dict for pools, with the file name as key, you cannot use that trick to remove the default pool for both versions and end up with orphaned www.conf files. How to resolve that?

**Comment by pather87** [comment]
To remove the default www.conf pool, you can add:
`
php:
  ng:
    fpm:
      pools:
        'www.conf':
          enabled: False
`

With the multiphp extension, this is not possible anymore, since to remove both pools of each version:

`
php:
  ng:
    version:
    - "7.2"
    - "7.3"
    fpm:
      pools:
        'www.conf':
          phpversion: "7.2"
          enabled: False
        'www.conf':
          phpversion: "7.3"
          enabled: False
`
This is obviously a duplicate key error. Maybe should add a switch in pillars to remove the default pool of both PHP versions

**Comment by n-rodriguez** [comment]
I've just tested it :  great work :+1:

**Comment by arthurlogilab** [comment]
@n-rodriguez great news, let's try to get this merged. @philpep should take a look at this merge request today to give us an extra opinion.

**Comment by myii** [comment]
@arthurlogilab Since starting this PR, this formula has been updated to use `semantic-release`.  In order to benefit from that upon merge, the commit messages need to be updated accordingly.  Please refer back to the [supplied documentation](https://github.com/saltstack-formulas/php-formula/blob/master/docs/CONTRIBUTING.rst#commit-message-formatting).

A few converted examples from this PR:

* `feat(php/ng): support the use of a list of php versions`
* `refactor(php/ng): don't iterate on string, make sure list is not string`
* `style(pillar.example): add line break`

**Comment by arthurlogilab** [comment]
@myii great news, I'm eager to work with this tool. Does that mean I should do some `git commit --amend` then `git push --force` on my fork so that the existing commits get replaced ?

**Comment by myii** [comment]
> @myii great news, I'm eager to work with this tool. Does that mean I should do some `git commit --amend` then `git push --force` on my fork so that the existing commits get replaced ?

@arthurlogilab Effectively, yes.  When there are a number of commits needing their messages modified, I prefer to use `git rebase <starting_point> -i`.  Then `git push -f` as you've mentioned.

**Comment by arthurlogilab** [comment]
@myii done, lets see what travis says.

**Comment by myii** [comment]
You cracked it the first time, great job @arthurlogilab!

**Comment by myii** [comment]
@n-rodriguez Doesn't this need to be merged before #183?
@arthurlogilab Are you waiting for a specific reviewer before this can be merged?

**Comment by arthurlogilab** [comment]
@myii not waiting for a specific reviewer, no. Thanks !

**Comment by myii** [comment]
Merged, thanks @arthurlogilab and to all others for their comments and reviews.

**Pull request merged**